### PR TITLE
Make forward_rptr work for assert_distinct

### PR DESCRIPTION
**Pull request by msullivan**
Update some of the related for tests, also.

### CODE REVIEW

File: tests/test_edgeql_for.py [hunk 1]
```diff
@@ -346,7 +342,7 @@ async def test_edgeql_for_in_computable_02b(self):
         await self.assert_query_result(
             r'''
                 SELECT User {
-                    select_deck := (
+                    select_deck := ((
```
Comment (elprans) [inline tests/test_edgeql_for.py:345]: This hunk seems to be unrelated.

Comment (msullivan) [reply tests/test_edgeql_for.py:345]: Yeah, though it's related in the sense that it's a follow-up to your multiplicity PR involving FOR

**Review by elprans — APPROVED** [review]
